util/pkg/vfs/s3fs.go

```diff
@@ -92,15 +92,37 @@ func (p *S3Path) Remove() error {
 
 	klog.V(8).Infof("removing file %s", p)
 
-	request := &s3.DeleteObjectInput{}
-	request.Bucket = aws.String(p.bucket)
-	request.Key = aws.String(p.key)
+	request := &s3.ListObjectVersionsInput{
+		Bucket: aws.String(p.bucket),
+		Prefix: aws.String(p.key),
+	}
 
-	_, err = client.DeleteObject(request)
+	response, err := client.ListObjectVersions(request)
 	if err != nil {
-		// TODO: Check for not-exists, return os.NotExist
+		return fmt.Errorf("error listing versions %s: %v", p, err)
```

Comment:
Would this path be taken if not-exists?

Reply:
This will find no results if path doesn't exist.

Reply:
You meant it may be used as way to retrurn `os.NotExist` if `response.Versions` is empty?
That may actually be an improvement and not call the API twice.